portal/i18next_gulpfile.js

```diff
@@ -0,0 +1,186 @@
+/*
+ * supported languages
+ */
+const languagesArray = ['en-US', 'en-AU'];
+/*
+ * namespace
+ */
+const nameSpace = "translation";
+
+/*
+ * helper function for writing file
+ */
+function save(target) {
+  return result => {
+    fs.writeFileSync(target, result);
+  };
+};
+
+/*
+ * extracting text from js/html files into json file
+ */
+gulp.task('i18next-extraction', function() {
+    console.log("extracting text and generate json file ...")
+    del(['translations/js/src/translation.json']);
+    return gulp.src(['static/**/*.{js,html}'])
+               .pipe(scanner({
+                    lngs: languagesArray, // supported languages
+                    keySeparator: "|",
+                    nsSeparator: "|",
+                    attr: {
+                        list: ['data-i18n'],
+                        extensions: ['.js', '.html', '.htm']
+                    },
+                    func: {
+                        list: ['i18next.t', 'i18n.t'],
+                        extensions: ['.js', '.jsx']
+                    }
+                    ,
+                    resource: {
+                        //the source path is relative to current working directory as specified in the destination folder
+                        savePath: './src/' + nameSpace + '.json'
+                    }
+                }))
+              .pipe(gulp.dest('translations/js'));
+});
+
+/*
+ * convert json to pot (the definition file) for translator's consumption
+ */
+gulp.task('i18nextConvertJSONToPOT', ['i18next-extraction'], function() {
+
+    const options = {/* you options here */}
+   /*
+    * converting json to pot
+    */
+    console.log("converting JSON to POT...");
+    del(['translations/js/src/translation.pot']);
+    i18nextConv.i18nextToPot('en', fs.readFileSync(translationSourceDir+nameSpace+".json"), options).then(save(translationSourceDir+nameSpace+".pot"));
+
+});
+
+/*
+ * combine newly created pot file to existing messages.pot file ???
+ * do we need this step??
+ */
+gulp.task('combineAllPotFiles', ['i18nextConvertJSONToPOT'], function() {
+    console.log("combine all pot files ...")
+    return gulp.src([translationSourceDir+nameSpace+".pot", 'translations/messages.pot'])
+          .pipe(concatPo('messages.pot'))
+          .pipe(gulp.dest('translations'));
+});
+
+/*
+ * converting po to json files
+ * note translating existing po file to json, which will be consumed by the front end
+ * this task assumes that:
+ *    1. text has been extracted from js file into JSON file
+ *    2. translated JSON into POT
+ *    3. merged new POT into main POT file [need to check about this step]
+ *    4. Po files have been returned from translator after uploading POT file from #3
+ */
+gulp.task('i18nextConvertPOToJSON', function() {
+  console.log("converting po to json ...")
+  const options = {/* you options here */}
+
+  del(['static/files/**/translation.json']);
+
+   /*
+    * translating po file to json for supported languages
+    */
```

Comment:
Do we need to rely on a maintained list of supported languages here? Or is it possible to iterate through each dir in `portal/translations/`, and if a PO file is found in that lang's subdir (or more accurately, in `portal/translations/<lang_name>/LC_MESSAGES`, we generate a corresponding JSON file?

I think the goal is to just be able to maintain our list of "supported" languages in smartling, and have the upload/download steps work for all languages found (for uploading, we just need one POT file for all the languages, and for downloading, we get a zip that contains PO files for all the supported translation languages found in the smartling project). At least, that's how we designed the existing smartling integration stuff.